pages/api/submit.ts

```diff
@@ -19,44 +38,86 @@ async function serverSubmit(req: NextApiRequest, res: NextApiResponse) {
   delete newSubmittedValues.formValues;
   delete existingValues.newCard;
 
-  let page;
+  // The default is the current page. If there are no errors, then this gets
+  // updated and we can proceed to the next page.
+  let page = currentPage;
   let queryValues = "";
+  let addresses;
+  let results;
+  let errors;
 
-  // TODO: run values through validation. After the new form field values
-  // are validated, then merge all together:
-  newSubmittedValues = { ...newSubmittedValues, ...existingValues };
-  for (const [key, value] of Object.entries(newSubmittedValues)) {
-    queryValues += `&${key}=${value}`;
-  }
-  // A switch statement so later on we can do form value validation on specific
-  // fields based on the current page.
   switch (currentPage) {
     case "personal":
-      page = "location";
+      // Validate the form values for this specific step.
+      errors = validatePersonalFormData({}, newSubmittedValues);
+      if (isEmpty(errors)) {
+        page = "location";
+      }
       break;
```

Comment:
This is the same idea for all pages. Run the submitted data through the basic validation function and if there are no errors, go to the next page. All the current form values get converted later.

Reply:
In this case, `validatePersonalFormData` validates the first name, last name, birthdate, and email since those are the fields on that page.